core/src/toga/window.py

```diff
@@ -245,8 +245,8 @@ def __init__(
         App.app.windows.add(self)
 
         # If content has been provided, set it
-        if content:
-            self.content = content
+        # Otherwise, set content to an empty Box widget
+        self.content = content if content else toga.Box()
```

Comment:
This has resulted in a few branch coverage failures. They could be fixed by removing the `if` check at each branch and assuming that `content` is never None. 

If we're committing to that invariant, then we should also update the `content` setter to interpret `None` in the same way as the constructor does. However, we're not sure whether this invariant is actually a good idea, as it prevents `content=None` from being round-tripped.

The presence of a default Box in the app's main window has obviously also complicated many of the tests, even if they didn't involve the main window directly.

Let's leave this PR open to discuss this question, and consider alternative approaches to the original issue in #2818.